pkg/monitor/monitor/monitor_manager.go

```diff
@@ -338,3 +353,32 @@ func (mm *MonitorManager) removeIngressIfExist(monitor *v1alpha1.TidbMonitor, na
 	}
 	return mm.typedControl.Delete(monitor, ingress)
 }
+
+func (mm *MonitorManager) patchTidbClusterStatus(tcRef *v1alpha1.TidbClusterRef, monitor *v1alpha1.TidbMonitor) error {
+	tc, err := mm.cli.PingcapV1alpha1().TidbClusters(tcRef.Namespace).Get(tcRef.Name, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	var mergePatch []byte
+	if tcRef != nil {
+		mergePatch, err = json.Marshal(map[string]interface{}{
+			"status": map[string]interface{}{
+				"monitor": map[string]interface{}{
+					"name":      monitor.Name,
+					"namespace": monitor.Namespace,
+				},
+			},
+		})
+	} else {
+		mergePatch, err = json.Marshal(map[string]interface{}{
+			"status": map[string]interface{}{
+				"monitor": nil,
+			},
+		})
+	}
+	if err != nil {
+		return err
+	}
+	_, err = mm.cli.PingcapV1alpha1().TidbClusters(tc.Namespace).Patch(tc.Name, types.MergePatchType, mergePatch)
```

Comment:
Now the tc is synced frequently, suggest adding some retry here.

Reply:
Retrying is not necessary here. If error happens, I tend to end this reconciliation and wait for the next round.